lib/client.js

```diff
@@ -1495,6 +1495,23 @@ function _resume (client, sync) {
   }
 }
 
+function shouldSendContentLength (contentLength, method) {
+  if (contentLength > 0) {
+    return true
+  }
+  // Many servers expect a Content-Length for these methods
+  if (method === 'POST' || method === 'PUT' || method === 'PATCH') {
+    return true
+  }
+  if (contentLength === 0) {
+    if (method === 'GET' || method === 'HEAD') {
+      return false
+    }
+    return true
+  }
+  return false
```

Comment:
```suggestion
	return contentLength !== 0 ? method !== 'GET' && method !== 'HEAD' : true
```

We just want to entirely exclude the `GET`, `HEAD`, `TRACE`, `OPTIONS`, and `CONNECT` methods, which do not expect a body. If we see a content length different than 0 and have any method other than them, we should attach it. wdyt?

Reply:
Agreed here, for future reference, we're interpreting RFC7230 and RFC7231

```
   A payload within a GET request message has no defined semantics;

...

   A user agent SHOULD NOT send a
   Content-Length header field when the request message does not contain
   a payload body and the method semantics do not anticipate such a
   body.

```